### PR TITLE
feat(acms-cam-13): implement get all customers (back-end)

### PR DESCRIPTION
**Pull request by Iantomasi**
link to jira tickets:
https://champlainsaintlambert.atlassian.net/browse/ACMS-13
https://champlainsaintlambert.atlassian.net/browse/ACMS-36

## Context
This pull request is to get a running version of the spring boot application updated into main, tested with a successful HTTP Get request for customer accounts in the back-end.

## Changes
- Added gradle and docker files
- Configured dependencies
- Added database schema
- Populated the customer accounts table with dummy test data

### CODE REVIEW

File: ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customeraccountsmanagementsubdomain/businesslayer/CustomerAccountServiceImpl.java [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
-public class CustomerAccountServiceImpl {
+import com.champlain.ateliermecaniquews.customeraccountsmanagementsubdomain.datalayer.CustomerAccountRepository;
+import com.champlain.ateliermecaniquews.customeraccountsmanagementsubdomain.datamapperlayer.CustomerAccountRequestMapper;
+import com.champlain.ateliermecaniquews.customeraccountsmanagementsubdomain.datamapperlayer.CustomerAccountResponseMapper;
+import com.champlain.ateliermecaniquews.customeraccountsmanagementsubdomain.presentationlayer.CustomerAccountResponseModel;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
```
Comment (CrisWillCarry) [inline ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customeraccountsmanagementsubdomain/businesslayer/CustomerAccountServiceImpl.java:11]: Might want to use the annotations that we've learned to make the code look cleaner ( example FullArgsConstructor) and declare the classes as final

File: ateliermecanique-ws/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM gradle:7.4 as builder
+WORKDIR /usr/src/app
+COPY src ./src
+COPY build.gradle .
+RUN ["gradle", "bootJar"]
+
+EXPOSE 8080
+
```
Comment (CrisWillCarry) [inline ateliermecanique-ws/Dockerfile:8]: We will have to figure out a way of also running our react app with docker!

File: ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customeraccountsmanagementsubdomain/presentationlayer/CustomerAccountRequestModel.java [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+@Value
+@Builder
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class CustomerAccountRequestModel {
+
+    private String customerFirstName;
+    private String customerLastName;
+    private String customerEmail;
+    private String customerPhoneNumber;
+
```
[the remark this answers is not in the document]
Comment (Iantomasi) [reply ateliermecanique-ws/src/main/java/com/champlain/ateliermecaniquews/customeraccountsmanagementsubdomain/presentationlayer/CustomerAccountRequestModel.java:17]: made some progress on hashing the password but we're going to need to discuss it as a team in next scrum meeting

File: ateliermecanique-ws/Dockerfile (hunk 2 above)
Comment (Iantomasi) [reply ateliermecanique-ws/Dockerfile:8]: yessir

**Pull request merged**